File: apps/ios/Milo Tales/Network/FeedbackAPI.swift
//
//  FeedbackAPI.swift
//  Milo Tales
//

import Foundation

nonisolated enum FeedbackCategory: String, Codable, CaseIterable, Identifiable {
    case bug
    case idea
    case praise
    case other

    var id: String { rawValue }

    var label: String {
        switch self {
        case .bug: "Bug"
        case .idea: "Idea"
        case .praise: "Praise"
        case .other: "Other"
        }
    }

    var symbol: String {
        switch self {
        case .bug: "ant.fill"
        case .idea: "lightbulb.fill"
        case .praise: "heart.fill"
        case .other: "ellipsis.circle.fill"
        }
    }
}

nonisolated struct CreateFeedbackRequest: Encodable {
    let category: FeedbackCategory
    let message: String
    let rating: Int?
}

nonisolated struct FeedbackResponse: Decodable {
    let id: String
    let category: FeedbackCategory
    let message: String
    let rating: Int?
    let createdAt: String
}

enum FeedbackAPI {
    static func send(_ request: CreateFeedbackRequest) async throws -> FeedbackResponse {
        try await APIClient.shared.post("/api/v1/feedback", body: request)
    }
}

File: apps/android/app/src/main/java/com/cortexlumora/lunastories/network/FeedbackAPI.kt
package com.cortexlumora.lunastories.network

import io.ktor.client.request.setBody
import io.ktor.http.ContentType
import io.ktor.http.HttpMethod
import io.ktor.http.contentType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

enum class FeedbackCategory {
    @SerialName("bug") bug,
    @SerialName("idea") idea,
    @SerialName("praise") praise,
    @SerialName("other") other,
}

@Serializable
data class CreateFeedbackRequest(
    val category: FeedbackCategory,
    val message: String,
    val rating: Int? = null,
)

@Serializable
data class FeedbackResponse(
    val id: String,
    val category: FeedbackCategory,
    val message: String,
    val rating: Int? = null,
    val createdAt: String,
)

object FeedbackAPI {
    suspend fun create(req: CreateFeedbackRequest): FeedbackResponse =
        APIClient.request("/feedback") {
            method = HttpMethod.Post
            contentType(ContentType.Application.Json)
            setBody(req)
        }
}
